### PR TITLE
chore: consolidate env and move helper functions to utils.go

### PR DESCRIPTION
**Pull request by akshaysngupta**
This PR aims at cleaning up main.go.
1. move helper functions to a different file.
1. add function consolidating the env variable; setup defaults and use azContext.

### CODE REVIEW

File: pkg/environment/environment.go [hunk 1]
```diff
@@ -110,13 +120,45 @@ type EnvVariables struct {
 	HostedOnUnderlay           bool
 }
 
-var portNumberValidator = regexp.MustCompile(`^[0-9]{4,5}$`)
-var boolValidator = regexp.MustCompile(`^(?i)(true|false)$`)
+// Consolidate sets values from azContext and provide defaults
```
Comment (draychev) [inline pkg/environment/environment.go]: `provided defaults`   (missing `d`)

File: pkg/azure/context.go [hunk 2]
```diff
@@ -40,5 +40,12 @@ func NewAzContext(path string) (*AzContext, error) {
 		return nil, err
 	}
 
+	if context.VNetResourceGroup == "" {
```
Comment (draychev) [inline pkg/azure/context.go]: `context` is a bit misleading -- I thought this was Go's context, but it turns out it was the Azure context... wish we would have named this something different for readability (or name collision)

File: cmd/appgw-ingress/main.go [hunk 3]
```diff
@@ -72,13 +64,22 @@ func main() {
 		glog.Fatal("Error parsing command line arguments:", err)
 	}
 
+	// Setup enviroment
```
Comment (draychev) [inline cmd/appgw-ingress/main.go]: This comment technically should be - fetching environment variables etc.  But I even wonder if it is necessary. Seems the line below is fairly self-explanatory ;)

File: pkg/environment/environment.go [hunk 4]
```diff
@@ -9,13 +9,17 @@ import (
...
 
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/azure"
 	"github.com/golang/glog"
 )
 
 const (
 	// AzContextLocationVarName is an environment variable name. This file is available on azure cluster.
 	AzContextLocationVarName = "AZURE_CONTEXT_LOCATION"
 
+	// AzureEnvironmentVarName is an evironment variable name. It represents the cloud in which the application is running.
```
Comment (draychev) [inline pkg/environment/environment.go]: It would be helpful to give a few examples of what this could be.

Comment (akshaysngupta) [reply pkg/environment/environment.go]: I will remove this. I was experimenting with AzureEnvironment. Will move this out to another PR.

File: pkg/azure/context.go (hunk 2 above)
Comment (akshaysngupta) [reply pkg/azure/context.go]: sure! will do in a subsequent PR.

**Pull request merged**